kube-dns panic in federation code

[full log](https://github.com/kubernetes/kubernetes/files/326757/kubedns.log.txt)
Run like so:

```yaml
    - args:
      - --domain=cluster.local.
      - --dns-port=10053
      - --federations=federation=kubernetes-federation.test
      image: gcr.io/google_containers/kubedns-amd64:1.3
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x40 pc=0x63e9a6]

goroutine 60 [running]:
panic(0x16ddd60, 0xc820014060)
        /usr/local/go/src/runtime/panic.go:481 +0x3e6
k8s.io/kubernetes/pkg/dns.(*KubeDNS).getClusterZone(0xc8200b2420, 0x0, 0x0, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/dns/dns.go:632 +0x86
k8s.io/kubernetes/pkg/dns.(*KubeDNS).federationRecords(0xc8200b2420, 0xc82031af00, 0x6, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/dns/dns.go:598 +0x1bf
k8s.io/kubernetes/pkg/dns.(*KubeDNS).Records(0xc8200b2420, 0xc82031e780, 0x47, 0xc82031e700, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/dns/dns.go:461 +0xc92
k8s.io/kubernetes/vendor/github.com/skynetservices/skydns/server.(*server).AddressRecords(0xc82000a940, 0xc82031e780, 0x47, 0xe0001001c, 0xc82031e780, 0x47, 0x0, 0x0, 0x0, 0x7fc300000200, ...)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/skynetservices/skydns/server/server.go:421 +0xd8
k8s.io/kubernetes/vendor/github.com/skynetservices/skydns/server.(*server).ServeDNS(0xc82000a940, 0x7fc3fc1eccd8, 0xc820293c80, 0xc820120360)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/skynetservices/skydns/server/server.go:361 +0x28ff
k8s.io/kubernetes/vendor/github.com/miekg/dns.(*ServeMux).ServeDNS(0xc8200154f0, 0x7fc3fc1eccd8, 0xc820293c80, 0xc820120360)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/miekg/dns/server.go:209 +0xbb
k8s.io/kubernetes/vendor/github.com/miekg/dns.(*Server).serve(0xc820416820, 0x7fc3fc21d2f0, 0xc82031daa0, 0x7fc3fc21d2c8, 0xc8200154f0, 0xc820116200, 0x58, 0x200, 0xc8200dc078, 0xc8201125c0, ...)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/miekg/dns/server.go:576 +0x7ef
created by k8s.io/kubernetes/vendor/github.com/miekg/dns.(*Server).serveUDP
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/miekg/dns/server.go:530 +0x3e1
```

@kubernetes/sig-cluster-federation

--- a/pkg/dns/dns.go
+++ b/pkg/dns/dns.go
@@ -125,6 +125,7 @@ func NewKubeDNS(client clientset.Interface, domain string, federations map[strin
 		domain:           domain,
 		cache:            NewTreeCache(),
 		cacheLock:        sync.RWMutex{},
+		nodesStore:       kcache.NewStore(kcache.MetaNamespaceKeyFunc),
 		reverseRecordMap: make(map[string]*skymsg.Service),
 		domainPath:       reverseArray(strings.Split(strings.TrimRight(domain, "."), ".")),
 		federations:      federations,

